catalyst-gateway-crates/c509-certificate/src/extensions/mod.rs

```diff
@@ -0,0 +1,176 @@
+//! This module contains the implementation of the Extensions field in the c509
+//! certificate. Please refer to the [c509-certificate](https://datatracker.ietf.org/doc/draft-ietf-cose-cbor-encoded-cert/09/) for more information.
+
+use alt_name::AltName;
+use minicbor::Encoder;
+
+use crate::cbor_encoder::CborEncoder;
+
+pub mod alt_name;
+
+/// Extensions field is an array of `Extension`, representing multiple extensions in a
+/// C509 certificate.
+pub type Extensions = Vec<Extension>;
+
+/// Represents an Extension in a C509 certificate.
+/// Each Extension includes its type, value, and a critical flag.
+///
+/// # Fields
+/// * `extension_type` - The type of the extension as defined in the `ExtensionRegistry`.
+/// * `extension_value` - The value of the extension, which can be different types of
+///   extension values.
+/// * `critical` - A boolean indicating whether the extension is critical.
+pub struct Extension {
+    pub extension_type: ExtensionRegistry,
+    pub extension_value: ExtensionValue,
+    pub critical: bool,
+}
+
+/// Currently implemented values for an extension.
+#[allow(unused)]
+pub enum ExtensionValue {
+    /// Alternative names for the issuer.
+    IssuerAltName(AltName),
+    /// Alternative names for the subject.
+    SubjectAltName(AltName),
+}
+
+impl CborEncoder for ExtensionValue {
+    /// Encodes a `ExtensionValue` using the provided CBOR encoder.
+    ///
+    /// # Arguments
+    ///
+    /// * `encoder` - A mutable reference to a `minicbor::Encoder` to encode the
+    ///   extensions into.
+    fn encode(&self, encoder: &mut Encoder<&mut Vec<u8>>) {
+        match self {
+            ExtensionValue::IssuerAltName(alt_name) => alt_name.encode(encoder),
+            ExtensionValue::SubjectAltName(alt_name) => alt_name.encode(encoder),
+        }
+    }
+}
+
+#[allow(unused)]
+/// Enum of possible C509 extension types.
+/// Please refer to the [c509-certificate](https://datatracker.ietf.org/doc/draft-ietf-cose-cbor-encoded-cert/09/) Section 9.4 C509 Extensions Registry for more information.
+pub enum ExtensionRegistry {
+    SubjectKeyIdentifier = 1,
+    KeyUsage = 2,
+    SubjectAltName = 3,
+    BasicConstraints = 4,
+    CRLDistributionPoints = 5,
+    CertificatePolicies = 6,
+    AuthorityKeyIdentifier = 7,
+    ExtendedKeyUsage = 8,
+    AuthorityInfoAccess = 9,
+    SignedCertificateTimestampList = 10,
+    SubjectDirectoryAttributes = 24,
+    IssuerAltName = 25,
+    NameConstraints = 26,
+    PolicyMappings = 27,
+    PolicyConstraints = 28,
+    FreshestCRL = 29,
+    InhibitAnyPolicy = 30,
+    SubjectInfoAccess = 31,
+    IPResources = 32,
+    ASResources = 33,
+    IPResourcesBlocksV2 = 34,
+    ASResourcesV2 = 35,
+    BiometricInfo = 36,
+    PrecertificateSigningCert = 37,
+    OCSPNoCheck = 38,
+    QualifiedCertificateStatements = 39,
+    SMIMECapabilities = 40,
+    TLSFeatures = 41,
+    ChallengePassword = 255,
+}
```

Comment:
I think we can support OID here easily.
PEN is just OID but from a different known base, so we can handle that in encoding.
```suggestion
    OID(OidRegistry),
}
```
Look at this crate: https://docs.rs/oid-registry/0.7.0/oid_registry/index.html
And: https://docs.rs/asn1-rs/0.6.1/asn1_rs/